### ISSUE TITLE
Get rid of Newtonsoft.Json in the SDK

### ISSUE THREAD

**Issue by chyyran** [SnowflakePowered/snowflake#573]
We want to move off of Newtonsoft.Json and switch completely to `System.Text.Json`. We aren't too intertwined with advanced Newtonsoft.Json functionality, so this should be a relatively simply switch.

~~* Write `System.Text.Json.Serialization.Converter<T>` for existing `JsonConverters`~~
~~* Write a converter for `ModuleDefinition`~~
~~* ThemeProvider requires a converter for `ThemeDefinition` but since this is a POCO we should be able to get away with the default converter.~~
~~* `StoneProvider` uses `JsonConvert.DeserializeAnonymousType` which may be tricky to get rid of~~
~~* `JsonPluginProperties` should become backed by an ExpandoObject~~

**Comment by chyyran** [comment]
Just waiting on a better story for input config serialization.

New plan is to keep Newtonsoft.Json in the SDK for convenience purposes, but still move to S.T.Json within the SDK.

**Comment by chyyran** [comment]
gonna just write a regular s.t.j deserializer for the current mappings format, we'll deal with the rest later.

**Comment by chyyran** [comment]
As of #823 Newtonsoft.Json has been removed from the dependencies SDK.
